Automattic/wp-calypso
# Framework: Implement AsyncLoad for code-splitting using Babel plugin

Supersedes #5356

This pull request is related to #5356, implementing `<AsyncLoad />`, but leveraging a new [Babel plugin `transform-wpcalypso-async`](https://github.com/Automattic/babel-plugin-transform-wpcalypso-async) to handle transformations of prop values under the hood to `require.ensure` or `require` depending on the current environment configuration.

The new transform plugin also exposes a lower-level `asyncRequire` function for general usage. See [plugin documentation](https://github.com/Automattic/babel-plugin-transform-wpcalypso-async) for more information.

**Testing Instructions:**

Navigate to the editor, verifying that the sharing accordion is displayed after visiting the page.
1. Navigate to the [post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Note that a Sharing accordion appears in the sidebar

**Follow-up Tasks:**

We should improve `<AsyncLoad />` to accept additional props to control when the module is required, so that we can load in response to specific interactions (e.g. hovering the accordion title).

cc @mtias @ehg @mcsf @seear @umurkontaci
